src/masternodes/masternodes.cpp

```diff
@@ -115,14 +115,16 @@ CMasternode::State CMasternode::GetState(int height) const
 {
     if (resignHeight == -1) { // enabled or pre-enabled
         // Special case for genesis block
-        if (creationHeight == 0 || height >= creationHeight + GetMnActivationDelay(height)) {
-            return State::ENABLED;
+        if (creationHeight == 0 || (height < Params().GetConsensus().EunosPayaHeight && height >= creationHeight + GetMnActivationDelay(height))
```

Comment:
Readability: What do you think about lifting the `Params().GetConsensus().EunosPayaHeight` to a var on the top of the fn?